interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java

```diff
@@ -586,6 +597,87 @@ public void sendsTimeoutHeader() {
     Assert.assertEquals(572000L, (long) requestHeadersCapture.get().get(ChannelImpl.TIMEOUT_KEY));
   }
 
+  @Test
+  public void deadlineNotExceeded() {
+    serverDelayMillis.set(0);
+    // warm up the channel and JVM
+    blockingStub.emptyCall(Empty.getDefaultInstance());
+    TestServiceGrpc.newBlockingStub(channel)
+        .configureNewStub()
+        .setTimeout(10, TimeUnit.MILLISECONDS)
+        .build().emptyCall(Empty.getDefaultInstance());
+  }
+
+  @Test(timeout = 10000)
+  public void deadlineExceeded() {
+    serverDelayMillis.set(20);
+    // warm up the channel and JVM
+    blockingStub.emptyCall(Empty.getDefaultInstance());
+    TestServiceGrpc.TestServiceBlockingStub stub = TestServiceGrpc.newBlockingStub(channel)
+        .configureNewStub()
+        .setTimeout(10, TimeUnit.MILLISECONDS)
+        .build();
+    try {
+      stub.emptyCall(Empty.getDefaultInstance());
+      fail("Expected deadline to be exceeded");
+    } catch (Throwable t) {
+      assertEquals(Status.DEADLINE_EXCEEDED, Status.fromThrowable(t));
+    }
+  }
+
+  @Test(timeout = 10000)
+  public void deadlineExceededServerStreaming() throws Exception {
+    serverDelayMillis.set(10); // applied to every message
+    // warm up the channel and JVM
+    blockingStub.emptyCall(Empty.getDefaultInstance());
+    TestServiceGrpc.TestService asyncStub = TestServiceGrpc.newStub(channel)
+        .configureNewStub()
+        .setTimeout(30, TimeUnit.MILLISECONDS)
+        .build();
+    StreamingOutputCallRequest request = StreamingOutputCallRequest.newBuilder()
+        .setResponseType(PayloadType.COMPRESSABLE)
+        .addResponseParameters(ResponseParameters.newBuilder()
+            .setSize(1))
+        .addResponseParameters(ResponseParameters.newBuilder()
+            .setSize(1))
+        .addResponseParameters(ResponseParameters.newBuilder()
+            .setSize(1))
+        .addResponseParameters(ResponseParameters.newBuilder()
+            .setSize(1))
+        .build();
+    StreamRecorder<StreamingOutputCallResponse> recorder = StreamRecorder.create();
+    asyncStub.streamingOutputCall(request, recorder);
+    recorder.awaitCompletion();
+    assertEquals(Status.DEADLINE_EXCEEDED, Status.fromThrowable(recorder.getError()));
+    assertNotEquals(0, recorder.getValues().size());
+  }
+
+  @Test(timeout = 10000)
+  public void deadlineAppliedAfterClientInterceptors() throws Exception {
```

Comment:
This test we can probably get rid of. In #574 we are swapping to using a deadline, so this test would no longer make sense. #574 is pretty far along, and I expect will be merged today or tomorrow (and yes, it will cause some conflicts with this one, or vise versa).